### ISSUE TITLE
Autotools build improvement 

### ISSUE THREAD

**Issue by mcuee** [avrdudes/avrdude#1664]
**1) libreadline status**
As of now, for auto-tools build, the presence of libreadline is not correct. It always says libreadline is not present even though libreadline is present and the final binary is linked to libreadline.

I tried to fix it last time but it caused issues with macOS terminal mode and I have to revert the following PR.
* https://github.com/avrdudes/avrdude/pull/1280
* #1296 

**2) to reduce the warnings:** 
* https://github.com/avrdudes/avrdude/issues/1665

Reference: many warnings when using autotools build. It is good to fix or at least reduce the warnings.
* https://github.com/avrdudes/avrdude/pull/1663#issuecomment-1928881496

**3) to fix NetBSD Autotools build** 

* https://github.com/avrdudes/avrdude/issues/1669
Right now the build is broken for NetBSD as the script can not detect the libraries.

```
Configuration summary:
----------------------
DON'T HAVE libusb
DON'T HAVE libusb_1_0
DON'T HAVE libftdi1
DON'T HAVE libftdi
DON'T HAVE libreadline
```

**4) to fix OpenBSD Autotools build** 
https://github.com/avrdudes/avrdude/issues/1670

Right now there is an error in the last step.
```
Warning: target ./libavrdude.a (prerequisite of: avrdude) does not have any command (BUG)
  CCLD     avrdude
cc: error: no such file or directory: './libavrdude.a'
*** Error 1 in . (Makefile:1043 'avrdude': @echo "  CCLD    " avrdude;/bin/sh ./libtool --silent --tag=CC    --mode=link cc -Wall -Wextra -W...)
*** Error 1 in . (Makefile:2820 'all-recursive': @fail=;  if (target_option=k; case ${target_option-} in  ?) ;;  *) echo "am__make_running_w...)
*** Error 2 in /home/mcuee/build/avrdude/src (Makefile:886 'all')
```

**Comment by stefanrueger** [comment]
@mcuee Thanks for proactively starting issues on things we'd like improved. Can we please use fewer issues: we can summarise what needs improved, eg, have *one* issue for improving autotools build (listing the 4 aspects to improve) and *one* issue for improving documentation. It is unlikely we attract a number of people working in parallel on these issues (and for the autotools this is definitely not advisable). Also the new issue on jtag2updi improvement is kinda non-descript. I'd remove that for the time being or make it specific.

Ideally we never grow over a page of issues and discussions so we don't lose sight over things on the second page

**Comment by mcuee** [comment]
@stefanrueger 

Done.

**Comment by mcuee** [comment]
@stefanrueger 

I have trimmed down the number of issues and discussions to one page. Feel free to close those issues/discussions you think can be closed.

**Comment by mcuee** [comment]
The following issue should be present in autotools build as well.
* https://github.com/avrdudes/avrdude/issues/1683

**Comment by ndim** [comment]
> The following issue should be present in autotools build as well.
> 
>     * [avrintel.h missing from installation #1683](https://github.com/avrdudes/avrdude/issues/1683)

I will reflect in autoconf whatever comes out of the discussion over there.

**Comment by mcuee** [comment]
Remove the bug label now that the remaining portions are for enhancement.

**Comment by mcuee** [comment]
I think we can close this issue.
